hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/CacheInfoCollector.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.util.phonehome;
+
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.instance.impl.Node;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+class CacheInfoCollector implements MetricsCollector {
+
+    @Override
+    public Map<String, String> computeMetrics(Node hazelcastNode) {
+        Map<String, String> cacheInfo = new HashMap<>(1);
+
+        Collection<DistributedObject> distributedObjects = hazelcastNode.hazelcastInstance.getDistributedObjects();
+        long countCacheWithWANReplication = distributedObjects.stream().filter(distributedObject -> {
```

Comment:
This lambda is a bit too lenghty (for readability reasons it is good to keep lambdas small, my personal advice is no more than 3 lines). A complex predicate like this one can be broken up into multiple filter() calls, like first you filter by service name, and in the second you filter by `config.wanReplicationRef`. Please make this change, then we will examine the code again and refactor in one more iteration.